signup-service/src/main/java/io/mosip/signup/services/RegistrationService.java

```diff
@@ -318,17 +321,23 @@ public RegistrationStatusResponse updatePassword(ResetPasswordRequest resetPassw
             log.error("Transaction {} : reset password failed with response {}", transactionId, restResponseWrapper);
             throw new SignUpException(ErrorConstants.RESET_PWD_FAILED);
         }
+    }
+
+    public RegistrationStatusResponse updatePassword(ResetPasswordRequest resetPasswordRequest,
+                                           String transactionId) throws SignUpException{
+        RegistrationTransaction transaction = this.getTransaction(resetPasswordRequest, transactionId);
+        RestRequestWrapper restRequest = this.getRestRequestWrapper(transaction, transactionId, resetPasswordRequest);
+
+        log.debug("Transaction {} : start reset password", transactionId);
+        this.validateResetPasswordRequest(restRequest, transactionId);
 
         transaction.getHandlesStatus().put(getHandleRequestId(transaction.getApplicationId(),
                 "phone", resetPasswordRequest.getIdentifier()), RegistrationStatus.PENDING);
         transaction.setRegistrationStatus(RegistrationStatus.PENDING);
         cacheUtilService.setStatusCheckTransaction(transactionId, transaction);
 
-        notificationHelper.sendSMSNotificationAsync(resetPasswordRequest.getIdentifier(), transaction.getLocale(),
-                        FORGOT_PASSWORD_SMS_NOTIFICATION_TEMPLATE_KEY, null)
-                .thenAccept(notificationResponseRestResponseWrapper ->
-                    log.debug(notificationLogging, notificationResponseRestResponseWrapper)
-                );
+        notificationHelper.sendSMSNotification(resetPasswordRequest.getIdentifier(), transaction.getLocale(),
```

Comment:
During reset-pwd, it should be ok to send the notification in a separate thread and not wait until the notification is sent out to complete and return the response.